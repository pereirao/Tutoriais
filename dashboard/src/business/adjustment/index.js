// @flow

export type Adjustment = {
  id: number,
  created_at: string, // rails timestamp
  amount: number,
  credit: boolean,
  reason: string,
  description: string
};

export * as adjustment_actions from './actions';
export * as adjustment_epics from './epics';
export * as adjustment_helpers from './helpers';
export { default as adjustmentReducer } from './reducer';
export { default as adjustment_selectors } from './selectors';
