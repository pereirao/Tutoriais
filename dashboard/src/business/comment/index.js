// @flow
export type Author = {
  email: string
};

export type Comment = {
  id: number,
  created_at: string, // rails timestamp
  note: string,
  author: Author,
  posted_as: string
};

export * as comment_actions from './actions';
export * as comment_epics from './epics';
export * as comment_helpers from './helpers';
export { default as commentReducer } from './reducer';
export { default as comment_selectors } from './selectors';
