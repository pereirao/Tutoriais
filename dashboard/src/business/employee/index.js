// @flow

type EmployeeRole = 'supplier' | 'driver';

export type Employee = {
  id: number,
  first_name: string,
  last_name: string,
  active: boolean,
  email: string,
  roles: Array<EmployeeRole>
};

export * as employee_epics from './epics';
export * as employee_actions from './actions';
export * as employee_helpers from './helpers';
export { default as employeeReducer } from './reducer';
export { default as employee_selectors } from './selectors';
