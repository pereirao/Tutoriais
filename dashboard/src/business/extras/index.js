export * from './extras.dux';
export * from './extras.sagas';
