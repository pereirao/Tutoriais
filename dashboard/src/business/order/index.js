// @flow

export type OrderTags = {
  gift: boolean,
  out_of_hours: boolean,
  vip: boolean,
  corporate: boolean,
  scheduled: boolean,
  new_customer: boolean
};

export type BillingInfo = {
  name: string,
  card_type: string,
  card_last4: string
};

export type Address = {
  address1: string,
  address2: string,
  city: string,
  state: string,
  zip_code: string,
  coords: {lat: number, lng: number}
};

export type PickupDetail = {
  id: string,
  name: string,
  phone: string
};

export type Amounts = {
  total: number,
  subtotal: number,
  tax: number,
  tip: number,
  minibar_promos: number,
  store_discounts: number,
  discounts?: number, // FIXME: optional until sunset
  delivery_fee: number,
};

export type DeliveryService = {
  id: number,
  name: string,
  email: string,
  created_at: string,
  updated_at: string,
  minimal_delivery_time: number
}

export type DeliveryMethodType = 'on_demand' | 'shipped' | 'pickup';

export type DeliveryMethod = {
  type: DeliveryMethodType,
  expectation: string
};

export type AttributeFilters = 'vip' | 'gift' | 'scheduled' | 'exception';

export type OrderFilters = {
  date_range: {
    start: string, // iso-8601 timestamp
    end: string // iso-8601 timestamp
  },
  delivery_method_types: Array<DeliveryMethodType>,
  attributes: Array<AttributeFilters>
}

export type ExceptionMetadata = {
  order_item_sku?: string,
  replacement_item?: string,
  replacement_price?: string,
  correct_price?: string,
  incorrect_listing?: string,
  incomplete_address?: string,
  suspected_fraud?: string,
  failed_delivery?: string,
  driver_unavailable_why?: string,
  other?: string,
};

export type ExceptionType = 'out_of_stock' |
  'incorrect_pricing' |
  'incorrect_listing' |
  'incomplete_address' |
  'suspected_fraud' |
  'driver_unavailable' |
  'failed_delivery' |
  'other';

export type Exception = {
  type: ExceptionType,
  description: string,
  metadata: ExceptionMetadata
};

export const PROBLEM_TYPES = [
  'out_of_stock',
  'incorrect_pricing',
  'incorrect_listing',
  'incomplete_address',
  'suspected_fraud',
  'driver_unavailable',
  'failed_delivery',
  'other'
];

export type OrderListName = 'all' | 'active' | 'completed' | 'future';

export type RecipientInfo = { short_name: string, long_name: string, phone: string };

export type TimelineEntryType = 'created_at'
  | 'confirmed_at'
  | 'en_route_at'
  | 'delivered_at'
  | 'canceled_at'
  | 'comment';

export type TimelineEntry = {
  time: string,
  type: TimelineEntryType,
  meta?: { name?: string, driverName?: string }
}

export type OrderState = 'paid'
  | 'confirmed'
  | 'scheduled'
  | 'canceled'
  | 'exception'
  | 'en_route'
  | 'delivered';

export type OrderItem = {
  id: number,
  sku: string,
  name: string,
  item_size: string,
  unit_price: number,
  quantity: number
};

export type Order = {
  id: number,
  number: number,
  created_at: string, // rails timestamp
  confirmed_at: string, // rails timestamp
  canceled_at: string, // rails timestamp
  delivered_at: string, // rails timestamp
  scheduled_for: string, // rails timestamp
  scheduled_for_end: string, // rails timestamp
  order_time: string, // rails timestamp
  birthdate: string,
  notes: string,
  delivery_method: DeliveryMethod,
  state: OrderState,
  gift_message: string,
  recipient_info: RecipientInfo,
  customer_name: string,
  billing: BillingInfo,
  address?: Address,
  pickup_detail?: PickupDetail,
  amounts: Amounts,
  type_tags: OrderTags,
  receipt_url: string,
  order_items: Array<OrderItem>,
  comment_ids: Array<number>, // NOTE: shouldn't be hydrated
  order_adjustment_ids: Array<number>, // NOTE: shouldn't be hydrated
  driver_id: number,
  driver?: any,
  using_delivery_service: boolean,
  delivery_service: DeliveryService
};

export * as order_actions from './actions';
export * as order_helpers from './helpers';
export * as order_epics from './epics';
export { default as orderReducer } from './reducer';
export { default as order_selectors } from './selectors';
