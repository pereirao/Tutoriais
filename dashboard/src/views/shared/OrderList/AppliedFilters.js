// @flow

import React from 'react';
import _ from 'lodash';
import I18n from '../../../localization';
import MBButton from '../../elements/MBButton';
import MBLoader from '../../elements/MBLoader';
import {date_helpers} from '../../utils/helpers';
import style_sheet from './OrderList.module.css';

const {formatDateShort} = date_helpers;

// TODO: fix pills present logic and differ if from clear filters

type AppliedFiltersProps = {
  query: string,
  filters: Array<Object>,
  removeFilter: Object => void,
  clearFilters: () => void,
  total_count: number,
  is_fetching: boolean
};
const AppliedFilters = ({
                          query,
                          filters,
                          removeFilter,
                          clearFilters,
                          total_count,
                          is_fetching
                        }: AppliedFiltersProps) => {
  const pill_count = filters.length + (_.isEmpty(query) ? 0 : 1);
  return (
    <div className={style_sheet.appliedFiltersWrapper}>
      <ResultsCount key="results_count" visible={pill_count > 0} total_count={total_count}/>
      <AppliedFilterPill
        key="query_pill"
        value={query}
        type="query"
        removeFilter={removeFilter}
        is_fetching={is_fetching}/>
      {filters.map(filter => (
        <AppliedFilterPill
          key={`${filter.type}_${filter.value}_pill`}
          removeFilter={removeFilter}
          is_fetching={is_fetching}
          {...filter} />
      ))}
      <ClearFilters key="clear_filters" clearFilters={clearFilters} visible={pill_count > 1}/>
    </div>
  );
};

type AppliedFilterPillProps = {
  type: string,
  value: string,
  removeFilter: Object => void,
  is_fetching: boolean
};
const AppliedFilterPill = ({type, value, removeFilter, is_fetching}: AppliedFilterPillProps) => {
  if (_.isEmpty(value)) return null;
  let text = '';
  if (type === 'query') text = I18n.t('ui.body.filter_pill.query', {value});
  else if (type === 'start' || type === 'end') text = I18n.t(`ui.body.filter_pill.${type}`, {value: formatDateShort(value)});
  else text = I18n.t(`ui.body.filter_pill.${value}`);
  return (
    <div className={style_sheet.pill}>
      <p className={style_sheet.pillText}>{text}</p>
      {is_fetching ? <MBLoader type="spinner"/> : <RemoveIcon onClick={() => removeFilter(type, value)}/>}
    </div>
  );
};

const RemoveIcon = ({...div_props}) => (
  <div className={style_sheet.removeIconWrapper} {...div_props}>
    <svg className={style_sheet.removeIcon} height="16px" width="16px" version="1.1" xmlns="http://www.w3.org/2000/svg">
      <line x1="1" y1="11" x2="11" y2="1" stroke="#FFFFFF" strokeWidth="2"/>
      <line x1="1" y1="1" x2="11" y2="11" stroke="#FFFFFF" strokeWidth="2"/>
    </svg>
  </div>
);

const ClearFilters = ({clearFilters, visible}) => {
  if (!visible) return null;
  return (
    <MBButton button_type="secondary" size="small" onClick={() => clearFilters()}>
      {I18n.t('ui.button.clear_all')}
    </MBButton>
  );
};

const ResultsCount = ({total_count, visible}) => {
  if (!visible) return null;
  return (
    <p className={style_sheet.resultsCount}>
      {I18n.t('ui.body.results', {count: total_count})}
    </p>
  );
};

export default AppliedFilters;
