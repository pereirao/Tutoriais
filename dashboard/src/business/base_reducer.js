//@flow

import {combineReducers} from 'redux';
import {adjustmentReducer} from './adjustment';
import {commentReducer} from './comment';
import {employeeReducer} from './employee';
import {notificationReducer} from './notification';
import {notificationMethodReducer} from './notification_method';
import {orderReducer} from './order';
import {reportReducer} from './report';
import {uiReducer} from './ui';
import {sessionReducer} from './session';
import {variantReducer} from './variant';
import {settingsReducer} from './settings';
import substituteReducer from './substitute/substitute.dux';
import productReducer from './product/product.dux';
import {connectRouter} from "connected-react-router";

const appReducer = (history) => combineReducers({
  adjustment: adjustmentReducer,
  comment: commentReducer,
  employee: employeeReducer,
  notification: notificationReducer,
  notification_method: notificationMethodReducer,
  order: orderReducer,
  report: reportReducer,
  ui: uiReducer,
  router: connectRouter(history),
  session: sessionReducer,
  variant: variantReducer,
  settings: settingsReducer,
  substitute: substituteReducer,
  product: productReducer
});

const rootReducer = (history) => (state, action) => {
  let app_state = state;
  if (action.type === 'SESSION:RESET_STATE') {
    app_state = undefined;
  }
  return appReducer(history)(app_state, action);
};

export default rootReducer;
