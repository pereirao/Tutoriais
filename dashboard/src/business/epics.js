import _ from 'lodash';

import { combineEpics } from 'redux-observable';

import { adjustment_epics } from './adjustment';
import { comment_epics } from './comment';
import { order_epics } from './order';
import { session_epics } from './session';
import { employee_epics } from './employee';
import { notification_method_epics } from './notification_method';
import { report_epics } from './report';
import { variant_epics } from './variant';
import { settings_epics } from './settings';

// convert the imported epic objects into an array
const epics_by_module = [
  adjustment_epics,
  comment_epics,
  order_epics,
  session_epics,
  employee_epics,
  notification_method_epics,
  report_epics,
  variant_epics,
  settings_epics
];

const epics = _.flatMap(epics_by_module, (module_epic_exports) => {
  // we pull the actual epic functions out of the exports we recieved from the each file
  const module_epics = _.values(module_epic_exports);
  return module_epics;
});

const rootEpic = combineEpics(...epics);
export default rootEpic;
