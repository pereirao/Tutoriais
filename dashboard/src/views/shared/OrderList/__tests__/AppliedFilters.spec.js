import React from 'react';

// NOTE: the named export isn't a connected component
import AppliedFilters from '../AppliedFilters';
import { order_helpers } from '../../../../business/order';

it('renders correctly', () => {
  const filters = {
    date_range: {start: 'fake_timestamp', end: 'other_fake_timestamp'},
    delivery_method_types: ['on_demand'],
    attributes: ['gift']
  };
  expect(
    shallow(<AppliedFilters query="query" filters={order_helpers.appliedFiltersArray(filters)} />)
  ).toMatchSnapshot();
});
