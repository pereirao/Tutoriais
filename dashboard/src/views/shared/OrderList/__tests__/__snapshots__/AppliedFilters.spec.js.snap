// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="appliedFiltersWrapper"
>
  <ResultsCount
    visible={true}
  />
  <AppliedFilterPill
    type="query"
    value="query"
  />
  <AppliedFilterPill
    type="start"
    value="fake_timestamp"
  />
  <AppliedFilterPill
    type="end"
    value="other_fake_timestamp"
  />
  <AppliedFilterPill
    type="filter"
    value="on_demand"
  />
  <AppliedFilterPill
    type="filter"
    value="gift"
  />
  <ClearFilters
    visible={true}
  />
</div>
`;
